org.eclipse.debug.examples.mixedmode/src/org/eclipse/debug/internal/examples/mixedmode/DoNothingMainTab.java

```diff
@@ -114,12 +114,14 @@ public void launchConfigurationChanged(ILaunchConfiguration configuration) {
 			Set<String> modes = configuration.getModes();
 			modes.add(getLaunchConfigurationDialog().getMode());
 			if (!fButton.isDisposed()) {
-				boolean sel = fButton.getSelection();
-				if (!sel & modes.containsAll(getModes())) {
-					fButton.setSelection(true);
-				} else if (sel & !modes.containsAll(getModes())) {
-					fButton.setSelection(false);
-				}
+				fButton.getDisplay().syncExec(() -> {
```

Comment:
Isn't there a risk of deadlock if this is expected to run also in UI Thread?

Reply:
> Isn't there a risk of deadlock if this is expected to run also in UI Thread?

Seems I need to revive https://github.com/eclipse-platform/eclipse.platform.swt/pull/67 what exactly would allow such tings like "run in the UI thread but I'm not sure if this **is** the UI thread" ;-)